build.gradle

```diff
@@ -372,8 +399,42 @@ task copyHtml(type: Copy) {
 // TODO: Temporary workaround for new exes / JRE bundling not quite being officially ready yet
 distZip.exclude("*.x*.exe")
 
+task unpackZipForNSIS(type: Copy) {
+    dependsOn distZip
+
+    def zipFile = file("${buildDir}/distributions/TerasologyLauncher.zip")
+    def tmpDir = file("${buildDir}/tmp/distNSIS")
+    from zipTree(zipFile)
+    into tmpDir
+}
+
+task renderNSISScript(type: Copy) {
```

Comment:
I named it "render" because this task doesn't actually compile the script (this is done by `distNSIS`, which depends on this one); `win-setup/win-setup.nsi` is a *template* and this tasks transforms it into an intermediate product, a valid NSIS script which is placed at `${buildDir}/tmp/distNSIS`, by "filling the holes" with the right parameters (language strings, version information, etc - see lines below), which is then compiled by the `distNSIS` task. This is accomplished using the [expand method of the copy task](https://docs.gradle.org/3.4.1/dsl/org.gradle.api.tasks.Copy.html).

So I used "render" which is commonly used with templates, but I can change it if you want.